repair_order_group/models/repair_order.py

```diff
@@ -0,0 +1,180 @@
+from odoo.exceptions import UserError
+
+
+class RepairOrder(models.Model):
+    """Extend repair.order with grouping, cascade, and shared quotation logic.
+
+    Features added:
+      • Link repairs into logical groups (via `repair.order.group`)
+      • Automatically create groups when adding another repair
+      • Synchronize customer (partner) across all repairs in the same group
+      • Cascade confirmation/cancellation actions within the group
+      • Create a single sale order for all repairs of one group
+    """
+
+    _inherit = "repair.order"
+
+    group_id = fields.Many2one(
+        "repair.order.group",
+        string="Repair Group",
+        ondelete="set null",
+        check_company=True,
+        index=True,
+        help="Repairs sharing the same group can be processed and quoted together.",
+    )
+
+    grouped_repair_ids = fields.One2many(
+        related="group_id.repair_ids",
+        string="Grouped Repairs",
+        readonly=True,
+        help="All other repairs that belong to the same group.",
+    )
+
+    def action_add_another_repair(self):
+        """Create a new empty repair linked to the same group.
+
+        If the current repair is not yet assigned to a group, a new
+        `repair.order.group` is automatically created first.
+        The new repair will reuse basic logistics fields such as
+        partner, company, and locations, but not copy any parts or products.
+        """
+        self.ensure_one()
+        if not self.group_id:
+            self.group_id = self.env["repair.order.group"].create(
+                {
+                    "partner_id": self.partner_id.id,
+                    "company_id": self.company_id.id,
+                }
+            )
+
+        # Create new empty repair record within the same group
+        new_repair = self.env["repair.order"].create(
+            {
+                "group_id": self.group_id.id,
+                "partner_id": self.partner_id.id,
+                "company_id": self.company_id.id,
+                "picking_type_id": self.picking_type_id.id,
+                "location_id": self.location_id.id,
+                "location_dest_id": self.location_dest_id.id,
+            }
+        )
+
+        return {
+            "type": "ir.actions.act_window",
+            "res_model": "repair.order",
+            "res_id": new_repair.id,
+            "view_mode": "form",
+            "target": "current",
+        }
+
+    def write(self, vals):
+        """Propagate partner changes across all repairs in the same group.
+
+        When a user changes the partner on one repair, all other repairs
+        of the same group automatically receive the same partner, ensuring
+        data consistency. Recursion is prevented via the context flag
+        `skip_group_sync`.
+        """
+        res = super().write(vals)
+        if "partner_id" in vals and not self.env.context.get("skip_group_sync"):
+            for rec in self.filtered("group_id"):
```

Comment:
`filtered` will iterate the same recordset you are already iterating  another time. 
so you can simply do
```python
for rec in self:
    if not rec.group_id:
        continue
....
```